pulp_rpm/tests/performance/test_sync.py

```diff
@@ -152,3 +167,108 @@ def test_centos8_kickstart_baseos_on_demand(self):
+        delete_orphans(cls.cfg)
+        # Certificates processing
+        cls.cdn_client_cert = False
+        if os.environ['CDN_CLIENT_CERT'] \
+                and os.environ['CDN_CLIENT_KEY'] \
+                and os.environ['CDN_CA_CERT']:
+            # strings have escaped newlines from environmental variable
+            cls.cdn_client_cert = os.environ['CDN_CLIENT_CERT'].replace('\\n', '\n')
+            cls.cdn_client_key = os.environ['CDN_CLIENT_KEY'].replace('\\n', '\n')
+            cls.cdn_ca_cert = os.environ['CDN_CA_CERT'].replace('\\n', '\n')
+
+    @skip_if(bool, "cdn_client_cert", False)
+    def test_sync_with_certificate(self):
+        """Test sync against CDN.
+
+        1. create repository, appstream remote and sync
+            - remote using certificates and tls validation
+        2. create repository, baseos remote and sync
+            - remote using certificates without tls validation
+        3. Check both repositories were synced and both have its own 'productid' content
+            - this test covering checking same repo metadata files with different relative paths
+        """
+        # 1. create repo, remote and sync them
+        repo_appstream = self.repo_api.create(gen_repo())
+        self.addCleanup(self.repo_api.delete, repo_appstream.pulp_href)
+
+        body = gen_rpm_remote(
+            url=RPM_CDN_APPSTREAM_URL,
+            client_cert=self.cdn_client_cert,
+            client_key=self.cdn_client_key,
+            ca_cert=self.cdn_ca_cert
+        )
+        appstream_remote = self.remote_api.create(body)
+        self.addCleanup(self.remote_api.delete, appstream_remote.pulp_href)
+
+        repository_sync_data = RpmRepositorySyncURL(remote=appstream_remote.pulp_href)
+        sync_response = self.repo_api.sync(repo_appstream.pulp_href, repository_sync_data)
+        monitor_task(sync_response.task)
+
+        # 2. create remote and re-sync
+        repo_baseos = self.repo_api.create(gen_repo())
+        self.addCleanup(self.repo_api.delete, repo_baseos.pulp_href)
+
+        body = gen_rpm_remote(
+            url=RPM_CDN_BASEOS_URL, tls_validation=False,
+            client_cert=self.cdn_client_cert,
+            client_key=self.cdn_client_key,
+        )
+        baseos_remote = self.remote_api.create(body)
+        self.addCleanup(self.remote_api.delete, baseos_remote.pulp_href)
+
+        repository_sync_data = RpmRepositorySyncURL(remote=baseos_remote.pulp_href)
+        sync_response = self.repo_api.sync(repo_baseos.pulp_href, repository_sync_data)
+        monitor_task(sync_response.task)
+
+        # Get all 'productid' repo metadata files
+        productids = [productid for productid in self.repometadatafiles.list().results if
+                      productid.data_type == 'productid']
+
+        # Update repositories info
+        repo_baseos = self.repo_api.read(repo_baseos.pulp_href)
+        repo_appstream = self.repo_api.read(repo_appstream.pulp_href)
```

Comment:
Since you re-read it anyway, does it make sense to do a conversion to a dict once and not on every call?
E.g. `repo_baseos_dict = self.repo_api.read(repo_baseos.pulp_href).to_dict()`

Just a suggestion. I can merge it as is.

Reply:
I like the idea, so updated accordingly.